src/rrt/Tree.hpp

```diff
@@ -309,7 +382,13 @@ class Tree {
         }
 
         // Add a node to the tree for this state
-        _nodes.push_back(Node<T>(intermediateState, source));
+        Node<T> newNode =
+            Node<T>(intermediateState, source, _dimensions, _TToArray);
+        _kdtree.addPoints(
+            flann::Matrix<double>(newNode.coordinates(), 1, _dimensions));
+        _nodes.push_back(newNode);
```

Comment:
(coordinates also shouldn't really be a const function) since your giving a mutable pointer.